elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala

```diff
@@ -72,6 +72,24 @@ trait QueryDsl extends DslCommons with SortDsl {
       new QueryRoot(query, fromOpt, sizeOpt, sorts, timeout, sourceFilter)
     }
   }
+  
+  // This api should be incorporated into QueryRoot in 2.0 when breaking changes are allowed.
+  // Before that, this api can serve to extend QueryRoot with backward compatibility.
+  class ExtendedQueryRoot(override val query: Query,
```

Comment:
We have `QueryRoot` object which has an `apply` method that has parameters with default values. `QueryRoot` case class also has an `apply` method. 

Currently, all its parameters do not have default values, which is fine. 

However, if we were to add `terminateAfter: Option[Int]=None`, the code will not compile due to scala compiler disallowing overloaded methods with default arguments.